src/libtiled/customproperties.h

```diff
@@ -0,0 +1,142 @@
+
+/*
+ * customproperties.h
+ * Copyright 2011-2017, Thorbjørn Lindeijer <thorbjorn@lindeijer.nl>
+ *
+ * This file is part of libtiled.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE CONTRIBUTORS ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL THE CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <QString>
+#include <QStringList>
+#include <QColor>
+#include <QVector>
+#include <QMetaType>
+
+#include "properties.h"
+
+namespace Tiled {
+
+/**
+ * Temporary definition of a custom props type.
+ * It has a name, a color and a list of values;
+ */
+struct CustomProp
+{
+    CustomProp() : color(Qt::gray) {}
+
+    CustomProp(QString name,
+               const QColor &color)
+        : name(std::move(name))
+        , color(color)
+    {}
+
+    CustomProp(const CustomProp &prop) 
+    {
+        name  = prop.name;
+        color = prop.color;
+        values = prop.values;
+        currentIndex = 0;
+    }
+
+    ~CustomProp()
+    {}
+
+
+    QString name;
+
+    QColor color;
+
+    QStringList values;
+
+    int currentIndex;
```

Comment:
I don't think this belongs here. We can have a "default value" member of the custom type (though I'd say we pick the first one by default for now), but the actual value should not be stored along with the type.